addons/forum-time-zones/matcher.js

```diff
@@ -0,0 +1,64 @@
+export default async function ({ addon, global, console }) {
+  const twelveHourTime = addon.settings.get("twelve-hour-time") === true;
+  const boxHeads = document.querySelectorAll(".box-head");
+  boxHeads.forEach((el) => {
+    if (!el.getElementsByTagName("a")[0]) return;
+    const text = el.getElementsByTagName("a")[0].innerText;
+    const dateCheck = handleFormatting(text);
+    el.getElementsByTagName("a")[0].innerText = dateCheck;
+  });
+
+  // this part was mostly written by @towerofnix on github!
+
+  function handleFormatting(text) {
+    const formatter = new Intl.DateTimeFormat("default", {
+      hour12: twelveHourTime,
+      timeStyle: "medium",
+      dateStyle: "long",
+    });
+
+    const timeParts = text.replace(/\./g, "").match(/^(.*?)\s+([0-9][0-9]):([0-9][0-9]):([0-9][0-9])$/);
+    if (!timeParts) {
+      console.warn(`Failed to parse forum timestamp: ${text}`);
+      return;
+    }
+
+    const [day, hours, minutes, seconds] = timeParts.slice(1);
+
+    let date;
+
+    let isRelative = false;
+
+    switch (day) {
+      case "Today":
+        isRelative = true;
+        date = new Date();
+        break;
+      case "Yesterday":
+        isRelative = true;
+        date = new Date();
+        date.setDate(date.getDate() - 1);
+        break;
+      default:
+        date = new Date(day);
+    }
+
+    date.setHours(hours, minutes, seconds, 0);
+
+    if (isNaN(date.getTime())) {
+      console.warn(`Failed to calculate date/time for forum timestamp: ${text} - regex timeParts:`, timeParts);
+      return;
+    }
```

Comment:
This causes problems in other languages

Reply:
Its also not indented.

Reply:
> This causes problems in other languages

Good catch. Maybe we should reconsider whether it's actually necessary to add an option for relative dates? The forums already use relative dates for "today" and "yesterday".

Reply:
This relative dates option doesn't add anything other than "today" and "yesterday"! The main function for that option is to turn it off which is confusing

Reply:
Ohh, I see.

Reply:
Here's the solution: compare `addon.tab.scratchMessage("Yesterday")` instead of `"Yesterday"`. Same with `Today`.
We need to check if some language puts the time before the date.